### PR TITLE
rewrite calculation

### PR DESCRIPTION
**Pull request by JohannesMisch**

### CODE REVIEW

File: components/AddDeviceForm.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ export default function AddForm({ createDevice }) {
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
     createDevice(data);
+    console.log(data);
```
Comment (mbosselmann) [inline components/AddDeviceForm.js]: I found a console.log that should be removed. 👀

File: pages/index.js [hunk 2]
```diff
@@ -36,90 +41,159 @@ export default function Home({
...
-    .filter(filterByCategory("Work"))
-    .map(calculateDailyPowerConsumption)
-    .reduce(sum, 0);
+  const sums = calculateSums(devices);
 
-  const sumUpLighting = devices
-    .filter(filterByCategory("Lighting"))
-    .map(calculateDailyPowerConsumption)
-    .reduce(sum, 0);
+  console.log(sums);
```
Comment (mbosselmann) [inline pages/index.js]: I found a console.log that should be removed. 👀

File: pages/index.js [hunk 3]
```diff
@@ -36,90 +41,159 @@ export default function Home({
...
+          device.power_consumption_standby * (24 - device.average_usage_time);
+        //---------------------------------------------------------------locationStandby
+        accumulator.locationOverall[device.location] =
+          (accumulator.locationOverall[device.location] ?? 0) +
+          (device.power_consumption * device.average_usage_time +
+            device.power_consumption_standby *
+              (24 - device.average_usage_time));
+        //--------------------------------------------------------------locationOverall
+        return accumulator;
+      },
```
Comment (mbosselmann) [inline pages/index.js:88]: I think this might be a matter of taste, but I would prefer to have comments before the code it affects and not after. 🤔

**Review by mbosselmann — COMMENTED** [review]
Hi Johannes,
as we spoke about your calculation function today, there is nothing much to review here. Left three comments, but I will approve your changes and trust that you still take out the console.logs. ;)

Happy hacking! :octocat:

**Comment by JohannesMisch** [comment]
Thank you very much for your comments, I will implement these things

**Pull request merged**